Documentation/upgrade.md

```diff
@@ -0,0 +1,293 @@
+---
+title: Upgrading Rook
+weight: 40
+---
+
+# Upgrading the Rook Software
+This guide will walk you through the manual steps to upgrade the software in a Rook cluster from one version to the next.
+Rook is a distributed software system and therefore there are multiple components to individually upgrade in the sequence defined in this guide.
+After each component is upgraded, it is important to verify that the cluster returns to a healthy and fully functional state.
+
+
+This guide is just the beginning of upgrade support in Rook.
+The goal is to provide prescriptive guidance and knowledge on how to upgrade a live Rook cluster and we hope to get valuable feedback from the community that will be incorporated into an automated upgrade solution by the Rook operator.
+
+We welcome feedback and opening issues!
+
+## Considerations
+With this manual upgrade guide, there are a few notes to consider:
+* **WARNING:** Upgrading a Rook cluster is a manual process in its very early stages.  There may be unexpected issues or obstacles that damage the integrity and health of your storage cluster, including data loss.  Only proceed with this guide if you are comfortable with that.
+* Rook is still in an alpha state.  Migrations and general support for breaking changes across versions are not supported or covered in this guide.
```

Comment:
shall we point out that rook upgrades won't be supported until declared beta in v0.6? until then they would be at risk of breaking changes